core/src/test/scala/com/gu/tableversions/core/util/RichRefSpec.scala

```diff
@@ -0,0 +1,36 @@
+package com.gu.tableversions.core.util
+
+import cats.effect.IO
+import cats.effect.concurrent.Ref
+import org.scalatest.{FlatSpec, Matchers}
+
+class RichRefSpec extends FlatSpec with Matchers {
```

Comment:
Think this would be a good candidate for a `TableDriven` thingymajig.

And maybe calling to `.attempt` and asserting on `.isLeft` vs. `.isRight`?

Reply:
Perhaps. Can you think of more than the two cases though?

As for `.attempt`, is that to avoid having to catch the exception?

Reply:
Nope. `TableDriven` might be overkill then - how about just a couple of assertions in one test body?

`.attempt` was so that you could have a common type in a `Table` for expected results if you went `TableDriven`. But yeah, I'd probably reach for that before `the exception thrownBy` too.

Reply:
As discussed IRL, I played with this but decided to leave as-is.